File: src/main/java/testingil/webinar/tddapi/s12/Navigator.java
package testingil.webinar.tddapi.s12;

public class Navigator {

	private DistanceProvider distProvider;
	private Location startingPoint; 

	public Navigator(Location startingPoint, DistanceProvider distProvider) {
		this.distProvider = distProvider;
		this.startingPoint = startingPoint;
	}

	public Distance getDistanceFromDestination()	{
		return distProvider.getDistance(startingPoint, startingPoint);
	}

	public void setDestination(Location dest) {
	}

}

File: C#/TddApi/NavigationApp/s12.re_add_second_test/Navigator.cs
using System;

namespace NavigationApp.s12.re_add_second_test
{
    public class Navigator
    {
        DistanceProvider distProvider;
        Location startingPoint;

        public Navigator(Location loc, DistanceProvider distService)
        {
            distProvider = distService;
            startingPoint = loc;
        }

        public Distance getDistanceFromDestination()
        {
            return distProvider.getDistance(startingPoint, startingPoint);
        }

        public void setDestination(Location dest)
        {
                
        }
    }
}